libs/openFrameworks/graphics/ofTrueTypeFont.cpp

```diff
@@ -1162,8 +1164,16 @@ void ofTrueTypeFont::drawCharAsShape(uint32_t c, float x, float y, bool vFlipped
 
 //-----------------------------------------------------------
 float ofTrueTypeFont::stringWidth(const std::string& c) const{
-	ofRectangle rect = getStringBoundingBox(c, 0,0);
-	return rect.width;
+//	return getStringBoundingBox(c, 0,0).width;
+	int w = 0;
+	iterateString( c, 0, 0, false, [&]( uint32_t c, glm::vec2 pos ){
+		if ( c == '\t' ){
+			w += getGlyphProperties(' ').advance * spaceSize * TAB_WIDTH;
+		} else {
+			w += getGlyphProperties( c ).advance;
+		}
+	});
+	return w;
```

Comment:
Maybe this should be in `getStringBoundingBox`

Reply:
Yes I think the same. I'll take some time to test with right to left texts too, just to see if it works correctly

Reply:
Done. I didn't use the function itself so we dont' call twice iterateString

Reply:
Not sure I understand? `getStringBoundingBox` only has one iterateString. 

So couldn't the `stringWidth` function just do what it used to?

```
 	ofRectangle rect = getStringBoundingBox(c, 0,0);
 	return rect.width;
```

Sorry if I missed something?

Reply:
Thanks @dimitre - type stuff can be really tricky. 🙂
My first thought is including the xMin is okay - as people are requesting the bounding box representing the full bounds of the represented string. 

I could see us adding more utilities in the future so placing two strings side by side ( with different typefaces ) is easier ( I have run into this ). 

But I think negative values in some cases with 0,0 origin can be okay.

Reply:
So if we keep using xMin in bounding box it makes sense to keep stringWidth as a separate independent function, so we can measure only the character advance in stringWidth

Reply:
Ahh - okay, looking back at the original forum post I see the logic here. 👍
Let's do what you suggest and before the PR is merged we can do some tests, with [the case in the forum post](https://forum.openframeworks.cc/t/are-monospace-fonts-really-monospace/40358) and the existing examples.